From d623dc8545baf2f6eb4799785d32cdc79eeb4acb Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Thu, 7 Jul 2016 23:59:32 -0500
Subject: [PATCH 08/21] Starting IGA2 horizontal blank early by 1 count to
 avoid odd resolution

Subtracting 1 count from CrtcHBlankStart for IGA2 appears to
suppress some monitors from reporting horizontal resolution
that is 1 more than the desired horizontal resolution.
(i.e., 1601 rather than 1600)

Signed-off-by: Eric Kudzin <knoppix1337@yahoo.com>
Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 src/via_display.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/via_display.c b/src/via_display.c
index 9c2acc3..b49ed54 100644
--- a/src/via_display.c
+++ b/src/via_display.c
@@ -2194,10 +2194,13 @@ viaIGA2SetDisplayRegister(ScrnInfoPtr pScrn, DisplayModePtr mode)
 
 
     /* Set IGA2 horizontal blank start. */
-    /* Horizontal Blanking Start: 2048 - 1 (max) */
+    /* Subtracting 1 from CrtcHBlankStart appears to suppress some
+     * monitors from reporting horizontal resolution that is 1 more
+     * than the desired horizontal
+     * resolution. (i.e., 1601 rather than 1600) */
     DEBUG(xf86DrvMsg(pScrn->scrnIndex, X_INFO,
                         "IGA2 CrtcHBlankStart: %d\n", mode->CrtcHBlankStart));
-    temp = mode->CrtcHBlankStart;
+    temp = mode->CrtcHBlankStart - 1;
 
     /* 3X5.52[7:0] - Horizontal Blanking Start Bits [7:0] */
     hwp->writeCrtc(hwp, 0x52, temp & 0xFF);
-- 
1.9.1

